### PR TITLE
Fix #40941 - Suggest trigger at language boundaries

### LINKED ISSUE #40941: no property autocomplete in JSX(related with an empty space)
- VSCode Version: Version 1.19.1 (1.19.1) 0759f77bb8d86658bc935a10a64f6182c5a1eeba

- OS Version: macOS, 10.12.6

Steps to Reproduce:

TypeScript React, and https://twitter.com/jiyinyiyong/status/946642303950626817

Reproduces without extensions: Yes.

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #40941

**Bug**
Suggestions not auto triggered when you type `x` in `const z = <div x>`. See https://github.com/Microsoft/vscode/issues/40941#issuecomment-354886756 for explaination of root cause

**Proposed fix**
In the case where suggestions are triggered at the boundary between two languages, I believe we should trigger. This change adds special logic to handle that case by making sure we check the word at the  previous character position instead of at the current character positon when at language boundaries

### CODE REVIEW

File: src/vs/editor/contrib/suggest/suggestModel.ts [hunk 1]
```diff
@@ -46,7 +46,19 @@ export class LineContext {
 		}
 		const pos = editor.getPosition();
 		model.tokenizeIfCheap(pos.lineNumber);
-		const word = model.getWordAtPosition(pos);
+
+		// When we are at an embedded language boundary, check the word at the previous character
```
Comment (mjbvz) [inline src/vs/editor/contrib/suggest/suggestModel.ts]: This some pretty ugly logic. Let  me know if there is some better way to express this

**Review by jrieken — CHANGES_REQUESTED** [review]
Talked to @alexandrudima about this and the outcome is that `getWordAtPosition` should be fixed instead of the consumer. Today the function is biased towards the right but it should look left when (1) there was no result on the right and (2) when it started at a language-border.

**Comment by mjbvz** [comment]
Ok, I didn't do that initially because I was concerned about regressions. I've pushed an update that moves the fix into `getWordAtPosition`

Take a look at the test case for `getWordAtPosition` and embedded languages and let me know if you think the new behavior makes sense

**Comment by alexandrudima** [comment]
I fixed an issue where `LanguageConfigurationRegistry.getWordDefinition(languageId)` was used for the left biased word, but `languageId` was the language for the right biased word.

**Pull request merged**